File: std_vector/n_best_test.py
#!/usr/bin/env python
# encoding: utf-8

from n_best import pyfill_2d

if __name__ == '__main__':
    d2 = pyfill_2d()
    assert(d2[1][1] == 2)
    assert(d2[3][3] == 6)
    assert(d2[9][9] == 18)

File: std_vector/cpp-source/n_best_test.cpp
#include "n_best.h"
#include <assert.h>
#include <vector>
#include <stdlib.h>
#include <iostream>

int main() {
  std::vector<std::vector<int> > x; 
  nbest::fill_2d(x);
  
  assert(x[1][1] == 2);
  assert(x[3][3] == 6);
  assert(x[9][9] == 18);
  return 0;
}
